### PR TITLE
Adds infrastructure and actions for elements to send data to the worldtube

### PR DESCRIPTION
**Pull request by nikwit**
## Proposed changes

Adds an action for elements to  integrate and send data  to the worldtube `SendToWorldtube` and an action for the worldtube to receive that data and add up the different pieces of the integral `ReceiveElementData`.

~depends on #4780~

### CODE REVIEW

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/Inboxes.hpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#include "Time/TimeStepId.hpp"
+
+namespace CurvedScalarWave::Worldtube::Tags {
+
+/*!
+ * \brief Inbox of the worldtube singleton chare which receives quantities
+ * projected onto spherical harmonics.
+ *
+ * \details Each element abutting the worldtube projects both `Psi` and the time
+ * derivative of `Psi` onto spherical harmonics by integrating it over the
```
Comment (nikwit) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/Inboxes.hpp]: change to \f$ \Psi\f$

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/ElementActions/SendToWorldtube.hpp [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+struct SendToWorldtube {
+  static constexpr size_t Dim = 3;
+  using tags_to_send = tmpl::list<CurvedScalarWave::Tags::Psi,
+                                  ::Tags::dt<CurvedScalarWave::Tags::Psi>>;
+
+  using tags_to_slice_to_face = tmpl::list<
+      CurvedScalarWave::Tags::Psi, CurvedScalarWave::Tags::Pi,
+      CurvedScalarWave::Tags::Phi<Dim>,
+      gr::Tags::Shift<Dim, Frame::Inertial, DataVector>,
+      gr::Tags::Lapse<DataVector>, domain::Tags::Coordinates<Dim, Frame::Grid>,
```
Comment (nikwit) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/ElementActions/SendToWorldtube.hpp]: dont need coords

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/Inboxes.hpp (hunk 1 above)
Comment (nilsvu) [reply src/Evolution/Systems/CurvedScalarWave/Worldtube/Inboxes.hpp]: FYI you can also write just `$\Psi$` (we have a Doxygen filter that understands standard LaTeX)

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/ElementActions/SendToWorldtube.hpp [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+      const auto& element_id = db::get<domain::Tags::Element<Dim>>(box).id();
+      const auto& excision_sphere = db::get<Tags::ExcisionSphere<Dim>>(box);
+      const auto direction = excision_sphere.abutting_direction(element_id);
+      ASSERT(direction.has_value(), "Should be abutting");
+      const auto& mesh = db::get<domain::Tags::Mesh<Dim>>(box);
+      const auto face_mesh = mesh.slice_away(direction->dimension());
+      const size_t face_size = face_mesh.number_of_grid_points();
+
+      Variables<tmpl::flatten<
+          tmpl::list<tags_to_slice_to_face, ::Tags::TempScalar<0>>>>
```
Comment (nilsdeppe) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/ElementActions/SendToWorldtube.hpp]: [optional] You could also replace `flatten<...>` with `push_back<tags_to_slice_to_face, ::Tags::TempScalar<0>>`

File: src/Evolution/Systems/CurvedScalarWave/Worldtube/ElementActions/SendToWorldtube.hpp [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+                                  phi, l, m);
+          get(get<CurvedScalarWave::Tags::Psi>(Ylm_coefs)).at(index) =
+              definite_integral(psi_regular_times_det * spherical_harmonic,
+                                face_mesh);
+          get(get<::Tags::dt<CurvedScalarWave::Tags::Psi>>(Ylm_coefs))
+              .at(index) = definite_integral(
+              dt_psi_regular_times_det * spherical_harmonic, face_mesh);
+        }
+      }
+      ASSERT(index == num_modes, "indexing error");
```
Comment (nilsdeppe) [inline src/Evolution/Systems/CurvedScalarWave/Worldtube/ElementActions/SendToWorldtube.hpp]: Could you expand on the error message?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
LGTM! One small request and one optional. Please squash and rebase immediately!

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**